src/aro/target.zig

```diff
@@ -723,9 +725,189 @@ pub fn toLLVMTriple(target: std.Target, buf: []u8) []const u8 {
+
+            return .no;
+        },
+
+        else => {
+            switch (target.cpu.arch) {
+                .hexagon => {
+                    // CLANG_DEFAULT_PIE_ON_LINUX
+                    return if (target.os.tag == .linux or target.isAndroid() or target.isMusl()) .yes else .no;
+                },
+
+                else => return .no,
+            }
+        },
+    };
+}
+
+fn isPICdefault(target: std.Target) DefaultPIStatus {
+    return switch (target.os.tag) {
+        .aix,
+        .haiku,
+
+        .macos,
+        .ios,
+        .tvos,
+        .watchos,
+        .visionos,
+        .driverkit,
+
+        .amdhsa,
+        .amdpal,
+        .mesa3d,
+
+        .ps4,
+        .ps5,
+        => .yes,
+
+        .fuchsia,
+        .cuda,
+        .zos,
+        .shadermodel,
+        => .no,
+
+        .dragonfly,
+        .openbsd,
+        .netbsd,
+        .freebsd,
+        .solaris,
+        .hurd,
+        => {
+            return switch (target.cpu.arch) {
+                .mips64, .mips64el => .yes,
+                else => .no,
+            };
+        },
+
+        .linux,
+        .elfiamcu,
+        => {
+            if (target.abi == .ohos)
+                return .no;
+
+            return switch (target.cpu.arch) {
+                .mips64, .mips64el => .yes,
+                else => .no,
+            };
+        },
+
+        .windows => {
+            if (target.isMinGW())
+                return if (target.cpu.arch == .x86_64 or target.cpu.arch == .aarch64) .yes else .no;
+
+            if (target.abi == .itanium)
+                return if (target.cpu.arch == .x86_64) .yes else .no;
+
+            if (target.abi == .msvc or target.abi == .none)
+                return .depends_on_linker;
+
+            if (target.ofmt == .macho)
+                return .yes;
+
+            return switch (target.cpu.arch) {
+                .x86_64, .mips64, .mips64el => .yes,
+                else => .no,
+            };
+        },
+
+        else => {
+            if (target.ofmt == .macho)
+                return .yes;
+
+            return switch (target.cpu.arch) {
+                .mips64, .mips64el => .yes,
+                else => .no,
+            };
+        },
+    };
+}
 /// This currently just returns the desired settings without considering target defaults / requirements
-pub fn getPICMode(target: std.Target, desired_pic_level: ?backend.CodeGenOptions.PicLevel, wants_pie: ?bool) struct { backend.CodeGenOptions.PicLevel, bool } {
-    _ = target;
+pub fn getPICMode(
+    target: std.Target,
+    use_linker: ?[]const u8,
+    desired_pic_level: ?backend.CodeGenOptions.PicLevel,
+    wants_pie: ?bool,
+) struct { backend.CodeGenOptions.PicLevel, bool } {
+    const linker = use_linker orelse system_defaults.linker;
+    const is_pie = switch (isPIEDefault(target)) {
+        .yes => true,
+        .no => false,
+        .depends_on_linker => if (std.mem.eql(u8, linker, "bfd"))
```

Comment:
To match the clang code (I'm not sure how much that matters) we should use `std.ascii.startsWithIgnoreCase` here instead of `std.mem.eql`